set.mm

```diff
@@ -526542,71 +526560,316 @@ elements in a zero ring (see ~ 0ringirng ).  (Contributed by Thierry
+          FWMYOWRWMXBYEUIYOVBYOXAYDWSUHWRWMWTUGVCVDVEVFVGVHVIVIVJWPYLWMWMWSSWMU
+          GQZUHQZUIQZRZXOUEZCUNTZFWKTZEWKTZLWKTZKSWKASWKUDWNPVKZWTSRZYLUUDVAWPU
+          UFYJUUBLEWKWKUUFYHYTFCWKUNUUFYGYSXOUUFYFYRWMUUFYEYQWMUIUUFYDYPWSUHWTS
+          WMUGVLVDVDVFVGVIVIVJWPUUCYSWMSUGQZUOUAZXNRZUEZCUNTZFWKTZEWKTZLSWKUUEX
+          ESRZUUCUUMVAWPUUNUUAUUKEFWKWKUUNYTUUJCUNUUNXOUUIYSUUNXJUUHXNUUNXIUUGU
+          OXESWMUGVCVMVNVPVHVIVJWPUULYSUUHWMXLUGQZUOUAZRZUEZCUNTZFWKTZEWMWKYNXK
+          WMRZUULUUTVAWPUVAUUJUURFCWKUNUVAUUIUUQYSUVAXNUUPUUHUVAXMUUOUOXKWMXLUG
+          VLVMVFVPVIVJWPUUSYSUUHUUHRZUEZCUNTZFSWKUUEXLSRZUUSUVDVAWPUVEUURUVCCUN
+          UVEUUQUVBYSUVEUUPUUHUUHUVEUUOUUGUOXLSWMUGVCVMVFVPVHVJWPUVCWMWMSYPUHQZ
+          UIQZRZUVBUEZCSUNWPVOWSSRZUVCUVIVAWPUVJYSUVHUVBUVJYRUVGWMUVJYQUVFWMUIW
+          SSYPUHVLVDVFVGVJWPUVHUVBWPUVGWMSUIQWMWPUVFSWMUIWPYPWPSWMWPVQYCVRVSVDW
+          PWMYCVTWAWPUUHWBWCWDWDWDWDWDWDWFAWOWQYBUEVAWNABCDWKEFGWMHIJKLMNOWKWGW
+          EVKWHWIWJ $.
+      $}
 
+      $d C a b c e f n o s t x $.  $d C d n o s x $.  $d C m n o $.  $d N m $.
+      $d n o r s x $.
+      $( ` 0 ` and ` 1 ` are in all steps of the construction of constructible
+         points.  (Contributed by Thierry Arnoux, 25-Jun-2025.) $)
+      constr01 $p |- ( ph -> { 0 , 1 } C_ ( C ` N ) ) $=
+        ( wcel cfv wss c0 wceq vm vn vo con0 cc0 c1 cpr cv fveq2 sseq2d constr0
+        csuc eqimss2i wa simpr simpl c0ex prid1 a1i sseldd constrsslem sstrd ex
+        wlim wral ciun 0ellim eqtrdi ssiun2s syl cvv vex constrlim sseqtrrd a1d
+        id tfinds ) AGUDPUEUFUGZGDQZRZOVRUAUHZDQZRZVRSDQZRVRUBUHZDQZRZVRWEULZDQ
+        ZRZVTUAUBGWASTWBWDVRWASDUIUJWAWETWBWFVRWAWEDUIUJWAWHTWBWIVRWAWHDUIUJWAG
+        TWBVSVRWAGDUIUJWDVRBCDEFHIJKLMNUKZUMWEUDPZWGWJWLWGUNZVRWFWIWLWGUOZWMBCD
+        EFWEHIJKLMNWLWGUPWMVRWFUEWNUEVRPWMUEUFUQURUSUTVAVBVCWAVDZWCWGUBWAVEWOVR
+        UCWAUCUHZDQZVFZWBWOSWAPVRWRRWAVGUCWAWQSVRWPSTWQWDVRWPSDUIWKVHVIVJWOBCDE
+        FUCWAVKHIJKLMNWAVKPWOUAVLUSWOVPVMVNVOVQVJ $.
+
+      $d C a b c e f s t x $.  $d C d $.  $d N a b c e f t $.  $d N d s x $.
+      $d a b c e f ph s t x $.  $d r s x $.
+      $( Constructed points are in the next generation constructed points.
+         (Contributed by Thierry Arnoux, 25-Jun-2025.) $)
+      constrss $p |- ( ph -> ( C ` N ) C_ ( C ` suc N ) ) $=
+        ( cc0 c1 cpr cfv constr01 wcel c0ex prid1 a1i sseldd constrsslem ) ABCD
+        EFGHIJKLMNOAPQRZGDSPABCDEFGHIJKLMNOTPUGUAAPQUBUCUDUEUF $.
+    $}
+
+    ${
+      $d C a b c e f s t x $.  $d C d $.  $d N a b c e f s t x $.  $d N d $.
+      $d X a b c e f t $.  $d X d $.  $d X r s x $.  $d a b c d e f ph s x $.
+      $d a b c e f ph s t x $.
+      constrconj.1 $e |- ( ph -> N e. On ) $.
+      constrconj.2 $e |- ( ph -> X e. ( C ` N ) ) $.
+      $( If a point ` X ` of the complex plane is constructible, its conjugate
+         ` ( * `` X ) ` is also constructible, in two more steps.  (Contributed
```

Comment:
It would be interesting to get more information about these "two steps". Maybe these could be extracted as lemmas?

Reply:
One should probably note that it's possible to prove that both $z$ and $\bar{z}$ appear on the same step. The reason is that complex conjugation maps lines to lines and circles to circles, so if, say, $z$ is the intersection of lines $A$ and $B$, then $\bar{z}$ is the intersection of $\bar{A}$ and $\bar{B}$.

I've looked at the part in Stewart which deals with that (Thierry's screenshot is taken from p. 94), and it seems like we might need this stronger property.

Reply:
Yes! I've updated this proof following this suggestion in https://github.com/metamath/set.mm/pull/4916/commits/b7bec87212fc0bea69c93b1b446fd7b77d1bc8dc.

Reply:
> It would be interesting to get more information about these "two steps". Maybe these could be extracted as lemmas?

The first step was to construct $2Re(z)$ by intersecting a circle centered in $z$ passing through $0$ with the $(0, 1)$ line, and the second step was to intersect two circles, one centered at $0$ and the other centered at $2Re(z)$, and both passing through $z$. Their other intersection is $\bar{z}$.

But I will not keep that original version, the one proposed by @savask is stronger.

The new version goes by induction over the construction steps, showing that at each step if any complex number $z$ can be constructed, so can its conjugate $\bar{z}$. The induction step checks each 3 cases (line-line, line-circle and circle-circle intersections).